File: apps/admin-web/src/data/staffDirectory.ts
/** Staff directory (mock). Slugs used in URLs. */

export type StaffRecord = {
  slug: string;
  name: string;
  role: string;
  phone?: string;
};

export const STAFF_RECORDS: StaffRecord[] = [
  {
    slug: "thabo-mokoena",
    name: "Thabo Mokoena",
    role: "Site supervisor",
    phone: "+27 82 000 1001",
  },
  {
    slug: "nomsa-khumalo",
    name: "Nomsa Khumalo",
    role: "Security officer",
    phone: "+27 83 000 1002",
  },
  {
    slug: "david-van-der-berg",
    name: "David van der Berg",
    role: "Armed response",
    phone: "+27 84 000 1003",
  },
  {
    slug: "lerato-maseko",
    name: "Lerato Maseko",
    role: "Control room",
    phone: "+27 82 000 1004",
  },
  {
    slug: "pieter-botha",
    name: "Pieter Botha",
    role: "Security officer",
    phone: "+27 83 000 1005",
  },
  {
    slug: "zanele-dlamini",
    name: "Zanele Dlamini",
    role: "Team lead",
    phone: "+27 84 000 1006",
  },
  {
    slug: "kevin-naidoo",
    name: "Kevin Naidoo",
    role: "Patrol",
    phone: "+27 82 000 1007",
  },
  {
    slug: "ayanda-ntuli",
    name: "Ayanda Ntuli",
    role: "Security officer",
    phone: "+27 83 000 1008",
  },
  {
    slug: "michelle-fourie",
    name: "Michelle Fourie",
    role: "Site supervisor",
    phone: "+27 84 000 1009",
  },
];

const bySlug = new Map(STAFF_RECORDS.map((s) => [s.slug, s]));
const byNameLower = new Map(STAFF_RECORDS.map((s) => [s.name.trim().toLowerCase(), s]));

export function getStaffBySlug(slug: string): StaffRecord | undefined {
  return bySlug.get(slug);
}

/** Resolve dashboard / customer guard strings to staff rows (exact name match). */
export function getStaffByName(displayName: string): StaffRecord | undefined {
  return byNameLower.get(displayName.trim().toLowerCase());
}

File: apps/employee_app/lib/data/staff_directory.dart
/// Mirrors `apps/admin-web/src/data/staffDirectory.ts`.

final class StaffRecord {
  const StaffRecord({
    required this.slug,
    required this.name,
    required this.role,
    this.phone,
  });

  final String slug;
  final String name;
  final String role;
  final String? phone;
}

const List<StaffRecord> kStaffRecords = [
  StaffRecord(
    slug: 'thabo-mokoena',
    name: 'Thabo Mokoena',
    role: 'Site supervisor',
    phone: '+27 82 000 1001',
  ),
  StaffRecord(
    slug: 'nomsa-khumalo',
    name: 'Nomsa Khumalo',
    role: 'Security officer',
    phone: '+27 83 000 1002',
  ),
  StaffRecord(
    slug: 'david-van-der-berg',
    name: 'David van der Berg',
    role: 'Armed response',
    phone: '+27 84 000 1003',
  ),
  StaffRecord(
    slug: 'lerato-maseko',
    name: 'Lerato Maseko',
    role: 'Control room',
    phone: '+27 82 000 1004',
  ),
  StaffRecord(
    slug: 'pieter-botha',
    name: 'Pieter Botha',
    role: 'Security officer',
    phone: '+27 83 000 1005',
  ),
  StaffRecord(
    slug: 'zanele-dlamini',
    name: 'Zanele Dlamini',
    role: 'Team lead',
    phone: '+27 84 000 1006',
  ),
  StaffRecord(
    slug: 'kevin-naidoo',
    name: 'Kevin Naidoo',
    role: 'Patrol',
    phone: '+27 82 000 1007',
  ),
  StaffRecord(
    slug: 'ayanda-ntuli',
    name: 'Ayanda Ntuli',
    role: 'Security officer',
    phone: '+27 83 000 1008',
  ),
  StaffRecord(
    slug: 'michelle-fourie',
    name: 'Michelle Fourie',
    role: 'Site supervisor',
    phone: '+27 84 000 1009',
  ),
];

final Map<String, StaffRecord> kStaffBySlug = {
  for (final s in kStaffRecords) s.slug: s,
};

StaffRecord? staffBySlug(String slug) => kStaffBySlug[slug];

StaffRecord? staffByName(String displayName) {
  final key = displayName.trim().toLowerCase();
  for (final s in kStaffRecords) {
    if (s.name.toLowerCase() == key) return s;
  }
  return null;
}

/// Default signed-in demo profile until Supabase maps users → staff rows.
const String kDemoStaffSlug = 'nomsa-khumalo';
